core/coreapi/interface/options/unixfs.go

```diff
@@ -0,0 +1,204 @@
+
+		Inline:       false,
+		InlineLimit:  32,
+		RawLeaves:    false,
+		RawLeavesSet: false,
+
+		Chunker: "size-262144",
+		Layout:  BalancedLayout,
+
+		Pin:      false,
+		OnlyHash: false,
+		Local:    false,
+	}
+
+	for _, opt := range opts {
+		err := opt(options)
+		if err != nil {
+			return nil, cid.Prefix{}, err
+		}
+	}
+
+	// (hash != "sha2-256") -> CIDv1
+	if options.MhType != mh.SHA2_256 {
+		switch options.CidVersion {
+		case 0:
+			return nil, cid.Prefix{}, errors.New("CIDv0 only supports sha2-256")
+		case 1, -1:
+			options.CidVersion = 1
+		default:
+			return nil, cid.Prefix{}, fmt.Errorf("unknown CID version: %d", options.CidVersion)
+		}
+	} else {
+		if options.CidVersion < 0 {
+			// Default to CIDv0
+			options.CidVersion = 0
+		}
+	}
+
+	// cidV1 -> raw blocks (by default)
+	if options.CidVersion > 0 && !options.RawLeavesSet {
+		options.RawLeaves = true
+	}
+
+	prefix, err := dag.PrefixForCidVersion(options.CidVersion)
+	if err != nil {
+		return nil, cid.Prefix{}, err
+	}
+
+	prefix.MhType = options.MhType
+	prefix.MhLength = -1
+
+	return options, prefix, nil
+}
+
+type unixfsOpts struct{}
+
+var Unixfs unixfsOpts
+
+// CidVersion specifies which CID version to use. Defaults to 0 unless an option
+// that depends on CIDv1 is passed.
+func (unixfsOpts) CidVersion(version int) UnixfsAddOption {
+	return func(settings *UnixfsAddSettings) error {
+		settings.CidVersion = version
+		return nil
+	}
+}
+
+// Hash function to use. Implies CIDv1 if not set to sha2-256 (default).
+//
+// Table of functions is declared in https://github.com/multiformats/go-multihash/blob/master/multihash.go
+func (unixfsOpts) Hash(mhtype uint64) UnixfsAddOption {
+	return func(settings *UnixfsAddSettings) error {
+		settings.MhType = mhtype
+		return nil
+	}
+}
+
+// RawLeaves specifies whether to use raw blocks for leaves (data nodes with no
+// links) instead of wrapping them with unixfs structures.
+func (unixfsOpts) RawLeaves(enable bool) UnixfsAddOption {
+	return func(settings *UnixfsAddSettings) error {
+		settings.RawLeaves = enable
+		settings.RawLeavesSet = true
+		return nil
+	}
+}
+
+// Inline tells the adder to inline small blocks into CIDs
+func (unixfsOpts) Inline(enable bool) UnixfsAddOption {
+	return func(settings *UnixfsAddSettings) error {
+		settings.Inline = enable
+		return nil
+	}
+}
+
+// InlineLimit sets the amount of bytes below which blocks will be encoded
+// directly into CID instead of being stored and addressed by it's hash.
+// Specifying this option won't enable block inlining. For that use `Inline`
+// option. Default: 32 bytes
+//
+// Note that while there is no hard limit on the number of bytes, it should
+// be kept at a reasonably low value, like 64 bytes if you intend to display
+// these hashes. Larger values like 256 bytes will work fine, but may affect
+// de-duplication of smaller blocks.
+//
+// Setting this value too high may cause various problems, such as render some
+// blocks unfetchable
```

Comment:
I think we are saying too much here and at this point will just be confusing the user. I would just simplify this with
> Note that while there is no hard limit on the number of bytes, it should be kept at a reasonably low value, such as 64 and no more then 1k.  Setting this value too high may cause various problems, such as render some blocks unfetchable.

@Stebalien thoughts?